Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.mm

```diff
@@ -385,15 +390,57 @@
     return activePoint;
 }
 
+void RemoteScrollingCoordinatorProxyIOS::displayDidRefresh(PlatformDisplayID displayID)
+{
+#if ENABLE(THREADED_ANIMATION_RESOLUTION)
+    updateAnimations();
+#endif
+}
+
+RemoteLayerTreeDrawingAreaProxyIOS& RemoteScrollingCoordinatorProxyIOS::drawingAreaIOS() const
+{
+    auto* drawingArea = dynamicDowncast<RemoteLayerTreeDrawingAreaProxy>(webPageProxy().drawingArea());
+    ASSERT(drawingArea && drawingArea->isRemoteLayerTreeDrawingAreaProxyIOS());
+    return *static_cast<RemoteLayerTreeDrawingAreaProxyIOS*>(drawingArea);
+}
+
 #if ENABLE(THREADED_ANIMATION_RESOLUTION)
 void RemoteScrollingCoordinatorProxyIOS::animationsWereAddedToNode(RemoteLayerTreeNode& node)
 {
     m_animatedNodeLayerIDs.add(node.layerID());
+    drawingAreaIOS().scheduleDisplayRefreshCallbacksForAnimation();
 }
 
 void RemoteScrollingCoordinatorProxyIOS::animationsWereRemovedFromNode(RemoteLayerTreeNode& node)
 {
     m_animatedNodeLayerIDs.remove(node.layerID());
+    if (m_animatedNodeLayerIDs.isEmpty())
+        drawingAreaIOS().pauseDisplayRefreshCallbacksForAnimation();
+}
+
+void RemoteScrollingCoordinatorProxyIOS::updateAnimations()
+{
+    // FIXME: Rather than using 'now' at the point this is called, we
+    // should probably be using the timestamp of the (next?) display
+    // link update or vblank refresh.
+    auto now = MonotonicTime::now();
+
+    auto& layerTreeHost = drawingAreaIOS().remoteLayerTreeHost();
+
+    auto animatedNodeLayerIDs = std::exchange(m_animatedNodeLayerIDs, { });
+    for (auto animatedNodeLayerID : animatedNodeLayerIDs) {
+        auto* animatedNode = layerTreeHost.nodeForID(animatedNodeLayerID);
+        ASSERT(animatedNode);
```

Comment:
Debug only assertions when you'll crash on the next line aren't really adding anything.

Reply:
Will remove that one and the one two lines below as well.